babel/src/board.js

```diff
@@ -283,20 +288,100 @@ export class Board {
         let hasWideStyleMatch = _.some(_.map(flatChunks, hasMatchInSingleRow));
         return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
+    
+    hasMatchEvent() {
+        return Boolean(findMatches(this.orbs)[0]);
+    }
 
-    swap(swapOrbs) {
-        let [[x1, y1], [x2, y2]] = swapOrbs;
+    swap(swapOrbs, playerSwap = true) {
+        let [[row1, col1], [row2, col2]] = swapOrbs;
         let orbsBefore = _.cloneDeep(this.orbs);
-        this.orbs[x1][y1] = orbsBefore[x2][y2]
-        this.orbs[x2][y2] = orbsBefore[x1][y1]
+        this.orbs[row1][col1] = orbsBefore[row2][col2]
+        this.orbs[row2][col2] = orbsBefore[row1][col1]
 
         // undo the swap if it did not yeild a match
-        if (!findMatches(this.orbs)[0]) {
+        if (playerSwap && !this.hasMatchEvent()) {
             this.orbs = orbsBefore;
         };
     }
+    
+    
+    _unmatch(row, col, match, skipToRandom = false) {
+        let thisOrb = this.orbs[row][col];
+        let swapped = false;
+        let directions = _.shuffle(['up', 'down', 'left', 'right']);
+        for (let i = 0; i < 4; i++) {
+            if (skipToRandom) { break };
+            if (directions[i] === 'up' && !_.isUndefined(this.orbs[row - 1]) && this.orbs[row - 1][col] !== thisOrb) {
+                this.swap([[row, col], [row - 1, col]], false);
+                swapped = true;
+                break;
+            } else if (directions[i] === 'down' && !_.isUndefined(this.orbs[row + 1]) && this.orbs[row + 1][col] !== thisOrb) {
+                this.swap([[row, col], [row + 1, col]], false);
+                swapped = true;
+                break;
+            } else if (directions[i] === 'left' && !_.isUndefined(this.orbs[row][col - 1]) && this.orbs[row][col - 1] !== thisOrb) {
+                this.swap([[row, col], [row, col - 1]], false);
+                swapped = true;
+                break;
+            } else if (directions[i] === 'right' && !_.isUndefined(this.orbs[row][col + 1]) && this.orbs[row][col + 1] !== thisOrb) {
+                this.swap([[row, col], [row, col + 1]], false);
+                swapped = true;
+                break;
+            }
+        }
+        while (!swapped) {
+            let [randomRow, randomCol] = [_.random(this.height - 1), _.random(this.width - 1)];
+            if (!_.includes(match, [randomRow, randomCol]) && this.orbs[randomRow][randomCol] !== thisOrb) {
+                this.swap([[row, col], [randomRow, randomCol]], false);
+                swapped = true;
+            }
```

Comment:
I decided to catch that instance here and having it jump directly to a random swap.